### PR TITLE
Feature: Add basic integration testing 👌

### PR DESCRIPTION
**Pull request by bengreenier**
+ Adds UX integration testing using `puppeteer` to instrument Chrome
+ Adds backend integration testing using `node-fetch` and an access token obtained from `puppeteer`
+ Validates the following scenarios:
```
Integration Tests
    √ health endpoint should be active (34ms)
    √ ui should require login (94ms)
    √ ui should login (4095ms)
    Authenticated
      √ should query titles (559ms)
      √ should query people (549ms)
      √ GET /titles (16ms)
      √ GET /people (18ms)
      Backend titles
        √ GET /title (11ms)
        POST /title
          √ creates (11ms)
        DELETE /title
          √ deletes (10ms)
      Backend people
        √ GET /people (10ms)
        POST /people
          √ creates (13ms)
        DELETE /people
          √ deletes (8ms)
```
+ Uses common testing infrastructure from `ui` part of the project

- [x] Document clearly in it's new README
- [x] Document clearly in main README
- [x] Validate usage of generic css selectors + text over specific css selectors
- [x] Discuss and determine where in the src tree this should live
- [x] Determine CI pattern to use

### CODE REVIEW

File: integration-test-tool/src/integration-tests.ts [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+          isAdult: false,
+          startYear: 1337,
+          endYear: null,
+          runtimeMinutes: 1337,
+          genres: [
+            'test',
+            'integration',
+          ],
+        }
+        // tslint:enable:object-literal-sort-keys
```
Comment (mindlessroman) [inline integration-test-tool/src/integration-tests.ts]: is this a comment or a way to set a setting in typescript?

File: integration-test-tool/src/integration-tests.ts [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+        expect(status).toBe(200)
+        expect(data).toEqual(firstPerson)
+      })
+
+      describe('POST /people', () => {
+        // tslint:disable:object-literal-sort-keys
+        const obj = {
+          nconst: 'inttest01',
+          primaryName: 'integration user 01',
+          birthYear: 1337,
```
Comment (mindlessroman) [inline integration-test-tool/src/integration-tests.ts]: lol

File: integration-test-tool/src/integration-tests.ts [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+          // instead we ensure that each kv pair from the original object is in the created object
+          Object.keys(obj).forEach((k) => {
+            expect(data[k]).toEqual(obj[k])
+          })
+        })
+      })
+
+      describe('DELETE /people', () => {
+        // tslint:disable:object-literal-sort-keys
+        const obj = {
```
Comment (mindlessroman) [inline integration-test-tool/src/integration-tests.ts]: This is a title object and not a people object.

File: ui/src/components/App.tsx [hunk 4]
```diff
@@ -56,8 +56,9 @@ export class App extends React.Component {
                     // log in
                     const graphScopes = [appConfig.clientID]
                     await userAgentApp.loginPopup(graphScopes)
-                    accessToken = await userAgentApp.acquireTokenSilent(graphScopes,
-                        'https://login.microsoftonline.com/microsoft.onmicrosoft.com')
+                    accessToken = await userAgentApp.acquireTokenSilent(graphScopes)
+                        // TODO(bengreenier): ensure both personal and professional still work
```
Comment (mindlessroman) [inline ui/src/components/App.tsx]: Left in comment?

File: integration-test-tool/src/integration-tests.ts (hunk 1 above)
Comment (bengreenier) [reply integration-test-tool/src/integration-tests.ts]: it controls the linter

**Pull request merged**